pyvista/core/filters/image_data.py

```diff
@@ -1604,3 +1606,287 @@ def _update_and_get_output():
+        ...     extraction_mode='point_seeds', point_seeds=points
+        ... )
+        >>> pl = pv.Plotter()
+        >>> _ = pl.add_mesh(connected.threshold(0.5), show_edges=True)
+        >>> _ = pl.add_mesh(
+        ...     connected.threshold(0.5, invert=True),
+        ...     show_edges=True,
+        ...     opacity=0.5,
+        ... )
+        >>> pl.show()
+
+        """
+        # Get a copy of input to not overwrite data
+        input_mesh = self.copy()  # type: ignore[attr-defined]
+
+        if scalars is None:
+            set_default_active_scalars(input_mesh)
+        else:
+            input_mesh.set_active_scalars(scalars)
+        # Make sure we have point data (required by the filter)
+        field, scalars = input_mesh.active_scalars_info
+
+        if field == FieldAssociation.CELL:
+            # Convert to point data
+            input_mesh = input_mesh.cells_to_points(scalars=scalars)
+
+        # Set vtk algorithm
+        alg = _vtk.vtkImageConnectivityFilter()
+        alg.SetInputDataObject(input_mesh)
+        alg.SetInputArrayToProcess(
+            0,
+            0,
+            0,
+            field.value,
+            scalars,
+        )  # args: (idx, port, connection, field, name)
+
+        # Set the scalar range considered for connectivity
+        if scalar_range is None:
+            # Default to vtk default 0.5 to VTK_DOUBLE_MAX, nothing to do
+            # See https://vtk.org/doc/nightly/html/classvtkImageConnectivityFilter.html
+            pass
+        else:
+            if scalar_range == 'auto':
+                unique_scalars = np.unique(input_mesh.point_data[scalars])
+                scalar_range = [unique_scalars[1], unique_scalars[-1]]
+
+            elif not isinstance(scalar_range, (np.ndarray, Sequence)):
+                raise TypeError('Scalar range must be a numpy array, a sequence, "auto" or None.')
+
+            if len(scalar_range) != 2:
+                raise ValueError('Scalar range must have two elements defining the min and max.')
+            if scalar_range[0] > scalar_range[1]:  # type: ignore[operator]
+                raise ValueError(
+                    f'Lower value of scalar range {scalar_range[0]}'
+                    f' cannot be greater than the upper value {scalar_range[0]}'
+                )
+            alg.SetScalarRange(*scalar_range)
+
+        if extraction_mode == 'all':
+            alg.SetExtractionModeToAllRegions()
+        elif extraction_mode == 'largest':
+            alg.SetExtractionModeToLargestRegion()
+        elif extraction_mode == 'point_seeds':
+            if point_seeds is None:
+                raise ValueError(
+                    '`point_seeds` must be specified when `extraction_mode="point_seeds"`.',
+                )
+            elif isinstance(point_seeds, (Sequence, np.ndarray)):
+                # Ensure points are floats, see pyvista.core.utilities.points
+                point_seeds = np.array(point_seeds, dtype=float)
```

Comment:
Small optimization to use `asarray` to avoid making a copy if input is already an array
```suggestion
                point_seeds = np.asarray(point_seeds, dtype=float)
```